tests/unit/core/api/test_search.js

```diff
@@ -0,0 +1,252 @@
+/* global window */
+import config from 'config';
+
+import search from 'core/api/search';
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  CLIENT_APP_ANDROID,
+  CLIENT_APP_FIREFOX,
+} from 'core/constants';
+import { parsePage } from 'core/utils';
+import { dispatchSignInActions, fakeAddon } from 'tests/unit/amo/helpers';
+import {
+  createApiResponse,
+  unexpectedSuccess,
+  userAgents,
+} from 'tests/unit/helpers';
+
+
+describe(__filename, () => {
+  let api;
+  let mockWindow;
+  const apiHost = config.get('apiHost');
+
+  beforeEach(() => {
+    api = dispatchSignInActions({
+      clientApp: CLIENT_APP_FIREFOX,
+    }).store.getState().api;
+    mockWindow = sinon.mock(window);
+  });
+
+  function _search(extraArguments = {}) {
+    return search({
+      api,
+      auth: true,
+      filters: { page: parsePage(3), query: 'foo' },
+      ...extraArguments,
+    });
+  }
+
+  function mockResponse(responseProps = {}) {
+    return createApiResponse({
+      jsonData: {
+        results: [
+          { ...fakeAddon, slug: 'foo' },
+          { ...fakeAddon, slug: 'food' },
+          { ...fakeAddon, slug: 'football' },
+        ],
+      },
+      ...responseProps,
+    });
+  }
+
+  it('sets the lang, limit, page and query', () => {
+    // FIXME: This shouldn't fail if the args are in a different order.
+    mockWindow.expects('fetch')
+      .withArgs(`${apiHost}/api/v3/addons/search/?app=firefox&page=3&q=foo&lang=en-US`)
+      .once()
+      .returns(mockResponse());
+    return _search().then(() => mockWindow.verify());
+  });
+
+  it('sets appversion if Firefox version is 57 or above', () => {
+    api = dispatchSignInActions({
+      clientApp: CLIENT_APP_FIREFOX,
+      userAgent: userAgents.firefox[5],
+    }).store.getState().api;
+
+    mockWindow.expects('fetch')
+      .withArgs(`${apiHost}/api/v3/addons/search/?app=firefox&appversion=57.1&page=3&q=foo&lang=en-US`)
+      .once()
+      .returns(mockResponse());
+
+    return _search().then(() => mockWindow.verify());
+  });
+
+  // See: https://github.com/mozilla/addons-frontend/pull/2969#issuecomment-323551742
+  it('does not set appversion if version is below 57', () => {
+    api = dispatchSignInActions({
+      clientApp: CLIENT_APP_FIREFOX,
+      userAgent: userAgents.firefox[4],
```

Comment:
Same thing here. It would be easier to follow this test if there was an alias like `userAgent: firefoxESR52`